### PR TITLE
vmware velocloud sdwan module

### PR DESCRIPTION
**Pull request by kpdhulipala**
vmware velocloud sdwan module

### CODE REVIEW

**Review by rling-equinix — CHANGES_REQUESTED** [review]

File: modules/velocloud-sdwan/variables.tf [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    rootPassword   = string
+  })
+
+  validation {
+    condition     = length(var.vendor_configuration.activationKey) > 0
+    error_message = "Activation Key has to be a non empty string."
+  }
+
+  validation {
+    condition     = length(var.vendor_configuration.controllerFqdn) > 0
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf]: input is not IP address format. its 
image

File: modules/velocloud-sdwan/variables.tf [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  validation {
+    condition     = can(regex("^[A-Z]{2}$", var.metro_code))
+    error_message = "Valid metro code consists of two capital letters, i.e. SV, DC."
+  }
+}
+
+variable "project_id" {
+  description = "project_id"
+  type        = string
+}
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf:14]: fixed

File: modules/velocloud-sdwan/variables.tf [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  description = "Identifier of an ACL template that will be applied on a device"
+  type        = string
+  validation {
+    condition     = try(var.acl_template_id != null, true)
+    error_message = "Acl template is required."
+  }
+}
+
+variable "connectivity" {
+  description = "Parameter to identify internet access for device. Supported Values: INTERNET-ACCESS(default) or PRIVATE or INTERNET-ACCESS-WITH-PRVT-MGMT"
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf:111]: fixed

File: modules/velocloud-sdwan/variables.tf [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  type        = string
+  default     = "INTERNET-ACCESS"
+}
+
+variable "additional_bandwidth" {
+  description = "Additional internet bandwidth for a device"
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf:121]: fixed

File: modules/velocloud-sdwan/variables.tf [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  type        = number
+  validation {
+    condition     = var.additional_bandwidth == 0 || (var.additional_bandwidth >= 25 && var.additional_bandwidth <= 5001)
+    error_message = "Additional internet bandwidth should be between 25 and 5001 Mbps."
+  }
+}
+
+variable "interface_count" {
+  description = "Number of network interfaces on a device. If not specified, default number for a given device type will be used."
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf:131]: fixed

File: modules/c8000v/variables.tf [hunk 6]
```diff
@@ -145,7 +145,7 @@ variable "secondary" {
     error_message = "Key 'additional_bandwidth' has to be between 25 and 5001 Mbps."
   }
   validation {
-    condition     = !try(var.secondary.enabled, false) || try(var.secondary.acl_template_id != null, true)
+    condition     = !try(var.secondary.enabled, false) || try(var.secondary.acl_template_id != null, false)
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/c8000v/variables.tf:148]: addressed for secondary acl.

File: modules/velocloud-sdwan/variables.tf [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    condition     = var.secondary.enabled ? can(regex("^[A-Z]{2}$", var.secondary.metro_code)) : true
+    error_message = "Key 'metro_code' has to be defined for secondary device. Valid metro code consists of two capital letters, i.e. SV, DC."
+  }
+  validation {
+    condition     = try(var.secondary.additional_bandwidth >= 25 && var.secondary.additional_bandwidth <= 5001, true)
+    error_message = "Key 'additional_bandwidth' has to be between 25 and 2001 Mbps."
+  }
+
+  validation {
+    condition     = try(!var.secondary.enabled || (length(var.secondary.account_number) > 0 && var.secondary.account_number != null), true)
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf]: fixed

File: modules/velocloud-sdwan/variables.tf [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+    condition     = length(var.notifications) > 0
+    error_message = "Notification list cannot be empty."
+  }
+}
+
+variable "acl_template_id" {
+  description = "Identifier of an ACL template that will be applied on a device"
+  type        = string
+  validation {
+    condition     = try(var.acl_template_id != null, true)
```
[the remark this answers is not in the document]
Comment (kpdhulipala) [reply modules/velocloud-sdwan/variables.tf]: fixed

**Review by rling-equinix — APPROVED** [review]
lgtm

File: modules/velocloud-sdwan/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+| <a name="input_version"></a> [version](#input\_version)                                                                                                                                                                                                                                               | VNF image version. Please refer to [certified VNF versions](https://docs.equinix.com/en-us/Content/Interconnection/NE/user-guide/NE-certified-VNFs.htm) document for the supported versions. If this value is not passed most recent and stable version will be used by invoking [equinix_network_device_software](https://registry.terraform.io/providers/equinix/equinix/latest/docs/data-sources/equinix_network_device_software) data source | `string`                                                                      | n/a                                     |   yes    |
```
Comment (rling-equinix) [inline modules/velocloud-sdwan/README.md:55]: version_number is the name for the input attribute.

File: modules/velocloud-sdwan/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+| <a name="input_package_code"></a> [package\_code](#input\_package\_code) Use [equinix_network_device_software](https://registry.terraform.io/providers/equinix/equinix/latest/docs/data-sources/equinix_network_device_software) datasource with device type code to find the supported package codes | Device software package                                                                                                                                                                                                                                                                                                                                                                                                                          | `string`                                                                      | n/a                                     |   yes    |
```
Comment (rling-equinix) [inline modules/velocloud-sdwan/README.md:62]: Could you put the reference link in the description section?

**Review by rling-equinix — CHANGES_REQUESTED** [review]

**Review by schadalawada — APPROVED** [review]

File: modules/velocloud-sdwan/README.md (hunk 10 above)
Comment (kpdhulipala) [reply modules/velocloud-sdwan/README.md]: fixed

File: modules/velocloud-sdwan/README.md (hunk 9 above)
Comment (kpdhulipala) [reply modules/velocloud-sdwan/README.md]: fixed

**Review by rling-equinix — APPROVED** [review]

**Review by schadalawada — APPROVED** [review]

**Pull request merged**